diff --git a/Makefile.global.in b/Makefile.global.in
--- a/Makefile.global.in
+++ b/Makefile.global.in
@@ -62,7 +62,9 @@ override CFLAGS += @CFLAGS@ @CITUS_CFLAGS@
 override BITCODE_CFLAGS := $(BITCODE_CFLAGS) @CITUS_BITCODE_CFLAGS@
 ifneq ($(GIT_VERSION),)
     override CFLAGS += -DGIT_VERSION=\"$(GIT_VERSION)\"
 endif
-override CPPFLAGS := @CPPFLAGS@ @CITUS_CPPFLAGS@ -I '${citus_abs_top_srcdir}/src/include' -I'${citus_top_builddir}/src/include' $(CPPFLAGS)
+# Add safestringlib include path for the safe memory functions used by columnar
+safestringlib_includedir = ${citus_abs_top_srcdir}/vendor/safestringlib/include
+override CPPFLAGS := @CPPFLAGS@ @CITUS_CPPFLAGS@ -I '${citus_abs_top_srcdir}/src/include' -I'${citus_top_builddir}/src/include' -I'${safestringlib_includedir}' $(CPPFLAGS)
 override LDFLAGS += @LDFLAGS@ @CITUS_LDFLAGS@

 # Make sure to keep the $(CPPFLAGS) at the end so that overriden variables
